dev/vcpkg/init.sh

```diff
@@ -0,0 +1,50 @@
+#! /bin/bash
+
+set -e
+
+exec 3>&1 >&2
+
+SCRIPT_ROOT="$(realpath "$(dirname "$0")")"
+VCPKG_ROOT="$SCRIPT_ROOT/.vcpkg"
+VCPKG="$SCRIPT_ROOT/.vcpkg/vcpkg"
+VCPKG_TRIPLET=x64-linux-avx
+
+cd "$SCRIPT_ROOT"
+
+if [ ! -d "$VCPKG_ROOT" ] || [ -z "$(ls "$VCPKG_ROOT")" ]; then
+    git clone https://github.com/microsoft/vcpkg.git --branch 2023.04.15 "$VCPKG_ROOT"
```

Comment:
can we download the tag instead of git clone here?
https://github.com/microsoft/vcpkg/archive/refs/tags/2023.04.15.tar.gz

Reply:
No. vcpkg heavily depends on the git log.

Reply:
maybe we could add "--depth 1"?

Reply:
It is possible to shallow clone a very small history, but vcpkg will not able to install old versions of packages. For example, velox depend on fmt 8.0, which was committed to vcpkg in 2021. We need `git clone --shallow-since=2021...` for vcpkg to find that commit. However, we can overwrite an old fmt port like other packages in `ports/` folder. So that vcpkg doesnt need history commits.

BTW, a full vcpkg clone need download about 70MB. since=2021 30MB, depth=1 5MB.